pyogrio/tests/test_geopandas_io.py

```diff
@@ -23,25 +28,22 @@
     from geopandas.array import from_wkt
     from geopandas.testing import assert_geodataframe_equal
 
-    from shapely.geometry import Point
-except ImportError:
-    pass
-
-has_pyarrow = False
-try:
-    import pyarrow  # noqa
+    import shapely  # if geopandas is present, shapely is expected to be present
 
-    has_pyarrow = True
 except ImportError:
     pass
 
 
 pytest.importorskip("geopandas")
 
 
-# Note: this will also be false for GDAL < 3.4 when GEOS may be present but we
-# cannot verify it
-has_geos = __gdal_geos_version__ is not None
+use_arrow = pytest.mark.parametrize(
+    "use_arrow",
+    [
+        False,
+        pytest.param(True, marks=requires_arrow_api),
+    ],
+)
```

Comment:
You could also make this a parametrized fixture, and then the `@use_arrow` is not needed, just the `use_arrow` as argument to the test function (that's also a bit less explicit though)

```suggestion
@pytest.fixture(params=[False, pytest.param(True, marks=requires_arrow_api)])
def use_arrow(request):
    return request.param
```